### PR TITLE
team: make teamd log to syslog

### PR DESCRIPTION
**Pull request by bengal**
When teamd is started in foreground it logs to stderr and its output
appears in the journal as coming from NM, which is quite confusing.

Use the new TEAM_LOG_OUTPUT environment variable [1] to tell teamd to
log through syslog, unless NetworkManager is in debug mode.

[1] https://github.com/jpirko/libteam/commit/e47d5db53873

https://bugzilla.redhat.com/show_bug.cgi?id=1596917

### CODE REVIEW

File: src/devices/team/nm-device-team.c [hunk 1]
```diff
@@ -479,11 +480,25 @@ teamd_child_setup (gpointer user_data)
...
 
+static const char **
+teamd_env (void)
+{
+	const char **env = g_new0 (const char *, 2);
+
+	if (nm_config_get_is_debug (nm_config_get ()))
+		env[0] = "TEAM_LOG_OUTPUT=stderr";
+	else
+		env[0] = "TEAM_LOG_OUTPUT=syslog";
```
Comment (thom311) [inline src/devices/team/nm-device-team.c]: does teamd need no environment variables at all? I wonder if there are some important once that we need to pass on as well?

Comment (bengal) [reply src/devices/team/nm-device-team.c]: It doesn't need any environment variable; indeed at the moment when NetworkManager is spawned by systemd (and teamd inherits the environment) the only variables set are PATH, LANG and the various LC_*, which are not used by teamd.

Comment (thom311) [reply src/devices/team/nm-device-team.c]: ACK

**Comment by bengal** [comment]
Merged as 94ea3a5e1b14878807c6e664efe47f802707595d.

**Comment by thom311** [comment]
See also: https://bugzilla.redhat.com/show_bug.cgi?id=1647414

**Pull request closed without merging**